src/e2e/java/teammates/e2e/cases/e2e/StudentHomePageE2ETest.java

```diff
@@ -0,0 +1,141 @@
+    private StudentHomePageNew studentHome;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/StudentHomePageE2ETest.json");
+
+        // use the 1st student account injected for this test
+        String student1GoogleId = TestProperties.TEST_STUDENT1_ACCOUNT;
+        String student1Email = student1GoogleId + "@gmail.com";
+        testData.accounts.get("alice.tmms").googleId = student1GoogleId;
+        testData.accounts.get("alice.tmms").email = student1Email;
+
+        // This student's name is Amy Betsy and has a registered account but yet to join course
+        testData.students.get("alice.tmms@SHomeUiT.CS2104").email = student1Email;
+
+        testData.students.get("alice.tmms@SHomeUiT.CS1101").googleId = student1GoogleId;
+        testData.students.get("alice.tmms@SHomeUiT.CS1101").email = student1Email;
+        testData.students.get("alice.tmms@SHomeUiT.CS4215").googleId = student1GoogleId;
+        testData.students.get("alice.tmms@SHomeUiT.CS4215").email = student1Email;
+        testData.students.get("alice.tmms@SHomeUiT.CS4221").googleId = student1GoogleId;
+        testData.students.get("alice.tmms@SHomeUiT.CS4221").email = student1Email;
+
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void allTests() throws Exception {
+        testContentAndLogin();
+        testLinks();
+        testLinkAndContentAfterDelete();
+    }
+
+    private void testContentAndLogin() throws Exception {
+
+        ______TS("content: no courses, 'welcome stranger' message");
+
+        String unregUserId = TestProperties.TEST_UNREG_ACCOUNT;
+        String unregPassword = TestProperties.TEST_UNREG_PASSWORD;
+        BackDoor.deleteStudent(unregUserId);
+
+        logout();
+        studentHome = getHomePageNew().clickStudentLogin().loginAsStudent(unregUserId, unregPassword);
+
+        browser.waitForPageLoad();
+        assertTrue(studentHome.verifyStrangerMessage());
+
+        ______TS("login successfully");
+
+        logout();
+        studentHome = getHomePageNew().clickStudentLogin()
+                                   .loginAsStudent(TestProperties.TEST_STUDENT1_ACCOUNT,
+                                                   TestProperties.TEST_STUDENT1_PASSWORD);
+
+        browser.waitForPageLoad();
+        assertTrue(studentHome.verifyFirstPanelFeedbackSessionName("[SHomeUiT.CS1101]: Programming Methodology"));
+
+        ______TS("login successfully as admin (masquerade mode)");
+
+        AppUrl detailsPageUrl = createUrl(Const.WebPageURIs.STUDENT_HOME_PAGE)
+                .withUserId(testData.students.get("SHomeUiT.charlie.d@SHomeUiT.CS2104").googleId);
+
+        StudentHomePageNew studentHomePage = loginAdminToPageNew(detailsPageUrl, StudentHomePageNew.class);
+        browser.waitForPageLoad();
+
+        assertTrue(studentHomePage.verifyAdminPagePresent());
+    }
+
+    private void testLinks() {
```

Comment:
No need to test this anymore. The links are already tested as part of front-end unit test.